### PR TITLE
Add more descriptive error message when uri has no db name

### PR DESCRIPTION
**Pull request by chrisbroome**

### CODE REVIEW

File: src/clojure/monger/core.clj [hunk 1]
```diff
@@ -233,10 +233,13 @@
...
    If username and password are provided, performs authentication."
   [^String uri-string]
-  (let [uri  (MongoClientURI. uri-string)
-        conn (MongoClient. uri)
-        db   (.getDB conn (.getDatabase uri))]
-    {:conn conn :db db}))
+  (let [uri    (MongoClientURI. uri-string)
+        conn   (MongoClient. uri)
+        dbName (.getDatabase uri)]
+    (if (nil? dbName)
```
[the remark this answers is not in the document]
Comment (chrisbroome) [reply src/clojure/monger/core.clj]: Ah right. So just `(if dbName` then...

File: src/clojure/monger/core.clj [hunk 2]
```diff
@@ -233,10 +233,13 @@
...
+      (throw (Exception. "No database name specified in uri"))
```
[the remark this answers is not in the document]
Comment (chrisbroome) [reply src/clojure/monger/core.clj]: Thanks! I'm clearly not a Java programmer, lol. Is the error message text ok?

**Comment by chrisbroome** [comment]
Looks like the tests are failing in CI because even though the script install mongodb 4.0.4, for some reason `mongod --version` returns `3.4.10`.

**Comment by chrisbroome** [comment]
Finally got the CI builds to pass by changing the Ubuntu distro to `xenial` which ships with MongoDB 4.0. I also had to increase the wait timeout to 15 seconds, which isn't ideal.

**Comment by michaelklishin** [comment]
Thank you!

**Comment by michaelklishin** [comment]
The Travis changes are acceptable. I might move the sleep part to `.travis.yml` since I use the script locally as well but the rest is reasonable 👍

**Pull request merged**